analyzer/runtime/runtime.go

```diff
@@ -306,29 +305,27 @@ func (m *processor) queueDbUpdates(batch *storage.QueryBatch, data *BlockData) {
 		batch.Queue(queries.AddressPreimageInsert, addr, preimageData.ContextIdentifier, preimageData.ContextVersion, preimageData.Data)
 	}
 
-	// Insert EVM token addresses.
+	// Insert EVM token addresses, and possibly dead-reckon its mutable properties (total_supply and num_transfers).
+	// Implementing totalSupply() is optional for ERC721 contracts, so we have to maintain this dead-reckoned fallback.
 	for addr, possibleToken := range data.PossibleTokens {
 		totalSupplyChange := possibleToken.TotalSupplyChange.String()
 		numTransfersChange := possibleToken.NumTransfersChange
-		if possibleToken.Mutated {
-			batch.Queue(queries.RuntimeEVMTokenAnalysisMutateUpsert, m.runtime, addr, totalSupplyChange, numTransfersChange, data.Header.Round)
-		} else {
-			batch.Queue(queries.RuntimeEVMTokenAnalysisInsert, m.runtime, addr, totalSupplyChange, numTransfersChange, data.Header.Round)
-		}
-		// Dead reckon total_supply and num_transfers.
-		// Note that total_supply is optional for ERC721 contracts.
-		// If the evm_tokens analyzer is able to fetch the total supply from the node,
-		// it will supersede this, but implementing totalSupply() is optional for ERC721 contracts,
-		// so we have to maintain this dead-reckoned fallback.
-		if numTransfersChange != 0 || possibleToken.TotalSupplyChange.Cmp(&big.Int{}) != 0 {
-			batch.Queue(
-				queries.RuntimeEVMTokenDeltaUpdate,
-				m.runtime,
-				addr,
-				totalSupplyChange,
-				numTransfersChange,
-			)
+		lastMutateRound := uint64(0)
+		if possibleToken.Mutated || possibleToken.TotalSupplyChange.Cmp(&big.Int{}) != 0 || possibleToken.NumTransfersChange != 0 {
+			// One of the mutable-and-queriable-from-the-runtime properties has changed in this round; mark that in the DB.
+			// NOTE: If _only_ numTransfers (which we cannot query from the EVM) changed, there's seemingly little point in
+			//       requesting a re-download. But in practice, totalSupply will change without an explicit burn/mint event,
```

Comment:
No :(

I looked into this some, but I only have observations, no deep understanding. I tested everythin on the wROSE contract, `0x21c718c22d52d0f3a789b752d4c2fd5908a8a733`
- totalSupply changes in most rounds that emit an event (not necessarily burn or mint!) involving that address. Those rounds are NOT at all the same as the rounds with txs referencing wROSE as a standalone arg.
- of the two txs that I checked and that generate events involving wROSE, both are about yuzuswap; [example](https://explorer.emerald.oasis.dev/tx/0xf0e19a06a073d47057e7aabcc6573d7407fa9242f5941ad34ad3d82a5dcc87d3/token-transfers)
    - to make things more confusing, that tx involves some transfer of 5 wROSE, but after the round, the total supply of wROSE increases by 1 (or 0.1 or some other power of 10; not sure about decimals).
    
IDK I'm inclined to say that the YUZURouter contract changes the total supply of wROSE somehow (internal transactions?) and events should be generated but they're not. I'm far from confident though. In case it's of interest, and in case I resume with this at some point, I'm attaching "private lab notes style" snippets I used when looking into this.

<details>
<summary>Log of things I tried / helpful snippets</summary>

This to find all the forms of wROSE's address: oasis format, eth format in hex and base64:
```
indexer=# select * from chain.evm_tokens  where symbol ='wROSE';
 runtime |                 token_address                  | token_type |  token_name  | symbol | decimals |        total_supply         | num_transfers | last_mutate_round | last_download_round 
---------+------------------------------------------------+------------+--------------+--------+----------+-----------------------------+---------------+-------------------+---------------------
 emerald | oasis1qpgcp5jzlgk4hcenaj2x82rqk8rrve2keyuc8aaf |         20 | Wrapped ROSE | wROSE  |       18 | 134273688211406143160770252 |          2113 |           1004297 |             1004298
indexer=# select address_data FROM chain.address_preimages where address=('oasis1qpgcp5jzlgk4hcenaj2x82rqk8rrve2keyuc8aaf');
                address_data                
--------------------------------------------
 \x21c718c22d52d0f3a789b752d4c2fd5908a8a733
oasisindexer=> select encode('\x21c718c22d52d0f3a789b752d4c2fd5908a8a733'::bytea, 'base64');
            encode            
------------------------------
 IccYwi1S0POnibdS1ML9WQiopzM=
(1 row)
```

This to dump the total balance for ~1000 rounds; I just plopped this at the beginning of the evm_tokens analyzer:
```go
	for round := 1003301; round <= 1004297; round++ {
		var wrose []byte = make([]byte, 20)
		// parse hex string `s` into x
		hex.Decode(wrose, ([]byte)("21c718c22d52d0f3a789b752d4c2fd5908a8a733"))
		tokenData, err := evm.EVMDownloadNewToken(
			ctx,
			m.logger,
			m.source,
			uint64(round),
			wrose,
		)
		if err != nil {
			m.logger.Error("error downloading token", "err", err)
			return
		}
		m.logger.Info("wROSE total supply", "round", round, "totalSupply", tokenData.TotalSupply)
	}
	os.Exit(0)
```

The log output (round, totalSupply) for the first several rounds is (annotations manual):
```
[1003301,133550131463522080000000000]
[1003302,133550131463522080000000000]
[1003303,133550131463522080000000000]
[1003304,133550131463522080000000000]
[1003305,133550131463522080000000000]
[1003306,133550136463522090000000000] change! has wROSE event.
[1003307,133550136463522090000000000]
[1003308,133550136463522090000000000]
[1003309,133557178326858330000000000] change! has wROSE event.
[1003310,133563967671930200000000000] change! has wROSE event.
[1003311,133563967671930200000000000]
[1003312,133570740220382000000000000] change! has wROSE event.
[1003313,133570740220382000000000000]
[1003314,133577253255795830000000000] change! has wROSE event.
[1003315,133577253255795830000000000]
[1003316,133577239443979430000000000] change! has wROSE event.
[1003317,133583821956026310000000000] change! has wROSE event.
[1003318,133583821956026310000000000] NO change. has wROSE event.
[1003319,133583821956026310000000000]
[1003320,133590316151646200000000000] change! has wROSE event.
```

This to find relevant events and txs (used to generate above annotations):
```sql
-- select * from chain.runtime_events where (body::text like '%IccYwi1S0POnibdS1ML9WQiopzM%' or body::text like '%21c718c22%' or body::text like '%oasis1qpgcp5jzlgk4hcenaj2x82rqk8rrve2keyuc8aaf%' or related_accounts::text like '%oasis1qpgcp5jzlgk4hcenaj2x82rqk8rrve2keyuc8aaf%') and runtime='emerald' and round=1004297;
select * from chain.runtime_events where runtime='emerald' and round>=1003301 and round<=1004297 and body::text like '%IccYwi1S0POnibdS1ML9WQiopzM=%';
select round from chain.runtime_transactions where runtime='emerald' and round>=1003301 and round<=1004297 and body::text like '%IccYwi1S0POnibdS1ML9WQiopzM=%'
```

</details>

Reply:
thanks for the thorough investigation!